diff --git a/CMakeLists.txt b/CMakeLists.txt
index 13342e9..44461d5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -107,8 +107,8 @@ enable_testing()
 # but hack for now
 
 add_subdirectory(src)
-add_subdirectory(benchmark)
-add_subdirectory(examples)
+#add_subdirectory(benchmark)
+#add_subdirectory(examples)
 
 # Install the header files
 install(DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/include" 
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 65b1cc6..95cfa25 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -34,6 +34,6 @@ target_link_libraries(gobjfs_static ${LIBRARY_LIST})
 install(TARGETS gobjfs_shared LIBRARY DESTINATION lib)
 install(TARGETS gobjfs_static ARCHIVE DESTINATION lib)
 
-add_subdirectory(util)
-add_subdirectory(test)
+# add_subdirectory(util)
+# add_subdirectory(test)
 add_subdirectory(networkxio)
diff --git a/src/networkxio/CMakeLists.txt b/src/networkxio/CMakeLists.txt
index a8210b0..397025e 100644
--- a/src/networkxio/CMakeLists.txt
+++ b/src/networkxio/CMakeLists.txt
@@ -30,14 +30,14 @@ target_link_libraries(gobjfs_client_static ${LIBRARY_LIST})
 # install library under install_dir/lib
 install(TARGETS gobjfs_client_shared
   LIBRARY DESTINATION lib)
-install(TARGETS gobjfs_client_static 
+install(TARGETS gobjfs_client_static
   ARCHIVE DESTINATION lib)
 
 ###################
-# gobjfs_server 
+# gobjfs_server
 ###################
 
-set(SERVER_SOURCES 
+set(SERVER_SOURCES
   NetworkXioIOHandler.cpp
   NetworkXioServer.cpp
   gobjfs_server.cpp
@@ -61,4 +61,4 @@ install(TARGETS gobjfs_server_shared
 install(TARGETS gobjfs_server_static 
   ARCHIVE DESTINATION lib)
 
-add_subdirectory(test)
+#add_subdirectory(test)
diff --git a/src/util/CMakeLists.txt b/src/util/CMakeLists.txt
index bd2488b..43f2c6a 100644
--- a/src/util/CMakeLists.txt
+++ b/src/util/CMakeLists.txt
@@ -1 +1 @@
-ADD_SUBDIRECTORY(test)
+#ADD_SUBDIRECTORY(test)
