diff -rupN nfs-ganesha-2.1.0-0.1.1-Source/cmake/gitdesc_from_path.sh nfs-ganesha-2.1.0-0.1.1-Source.new/cmake/gitdesc_from_path.sh
--- nfs-ganesha-2.1.0-0.1.1-Source/cmake/gitdesc_from_path.sh	2014-06-20 10:27:57.000000000 +0300
+++ nfs-ganesha-2.1.0-0.1.1-Source.new/cmake/gitdesc_from_path.sh	2015-03-03 16:08:53.741346395 +0200
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 res=`echo $1 | sed -e 's/^.*_desc_//g'  | sed -e 's/-[0-9]*\.[0-9]*\.[0-9]*$//g'`
 
 if [ -z "$res" ] ; then
diff -rupN nfs-ganesha-2.1.0-0.1.1-Source/cmake/githead_from_path.sh nfs-ganesha-2.1.0-0.1.1-Source.new/cmake/githead_from_path.sh
--- nfs-ganesha-2.1.0-0.1.1-Source/cmake/githead_from_path.sh	2014-02-20 14:46:22.000000000 +0200
+++ nfs-ganesha-2.1.0-0.1.1-Source.new/cmake/githead_from_path.sh	2015-03-03 16:09:02.069346308 +0200
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 res=`echo $1 | sed -e 's/[_-]/\n/g' | grep git | sed -e 's/git//g'`
 
 if [ -z "$res" ] ; then
